algorithms/a3c/env_vizdoom.py

```diff
@@ -0,0 +1,379 @@
+
+        return reward
+
+    def get_agent_and_object_positions(self):
+        """Get agent and object positions based on the difficulty
+        of the environment.
+        """
+        object_x_coordinates = []
+        object_y_coordinates = []
+
+        if self.params.difficulty == 'easy':
+            # Agent location fixed in Easy.
+            agent_x_coordinate = 128
+            agent_y_coordinate = 512
+            agent_orientation = 0
+
+            # Candidate object locations are fixed in Easy.
+            object_x_coordinates = [EASY_ENV_OBJECT_X] * 5
+            for i in range(-2, 3):
+                object_y_coordinates.append(
+                    Y_OFFSET + MAP_SIZE_Y/2.0 + OBJECT_Y_DIST * i)
+
+        if self.params.difficulty == 'medium':
+            # Agent location fixed in Medium.
+            agent_x_coordinate = 128
+            agent_y_coordinate = 512
+            agent_orientation = 0
+
+            # Generate 5 candidate object locations.
+            for i in range(-2, 3):
+                object_x_coordinates.append(np.random.randint(
+                    MEDIUM_ENV_OBJECT_X_MIN, MEDIUM_ENV_OBJECT_X_MAX))
+                object_y_coordinates.append(
+                    Y_OFFSET + MAP_SIZE_Y/2.0 + OBJECT_Y_DIST * i)
+
+        if self.params.difficulty == 'hard':
+            # Generate 6 random locations: 1 for agent starting position
+            # and 5 for candidate objects.
+            random_locations = generate_points(HARD_ENV_OBJ_DIST_THRESHOLD,
+                                               MAP_SIZE_X - 2*MARGIN,
+                                               MAP_SIZE_Y - 2*MARGIN, 6)
+
+            agent_x_coordinate = random_locations[0][0] + X_OFFSET + MARGIN
+            agent_y_coordinate = random_locations[0][1] + Y_OFFSET + MARGIN
+            agent_orientation = np.random.randint(4)
+
+            for i in range(1, 6):
+                object_x_coordinates.append(
+                    random_locations[i][0] + X_OFFSET + MARGIN)
+                object_y_coordinates.append(
+                    random_locations[i][1] + Y_OFFSET + MARGIN)
+
+        return agent_x_coordinate, agent_y_coordinate, agent_orientation, \
+            object_x_coordinates, object_y_coordinates
+
+    def get_candidate_objects_superlative_instr(self, correct_object):
+        '''
+        Get any possible combination of objects
+        and give the maximum size
+        SIZE_THRESHOLD refers to the size in terms of number of pixels so that
+        atleast there is minimum size difference between two objects for
+        instructions with superlative terms (largest and smallest)
+        These sizes are stored in ../data/object_sizes.txt
+        '''
+
+        instr_contains_color = False
+        # instr_contains_color is set if the instruction contains the color
+        # attribute (e.g.) "Go to the largest green object".
+        # instr_contains_color is True if the instruction doesn't contain the
```

Comment:
What is this madness? instr_contains_color == True only if there is NO color instruction?

Reply:
Ok docstring error, the function looks good.
```suggestion
        # instr_contains_color is True if the instruction contains the
        # color attribute. (e.g.) "Go to the smallest green object"
```

Reply:
Also I know that you have fobia to docstrings with """ instead of #, but I would say the rule of thumb is: If your comments are 3 lines or bigger then """, else #. Moreover, since the comment happens just after the docstring, why don't you add this info to the docstring itself?

Reply:
ok